File: DokanyAPI/DokanyApp.BLL/DTO/OrderCreationDto.cs
using DokanyApp.BLL.Entities;
using System;

namespace DokanyApp.BLL.DTO
{
    public class OrderCreationDto
    {
        public DateTime CreationDate { get; set; }
        public DateTime ShippingDate { get; set; }
        public int ShippingId { get; set; }
        public string Description { get; set; }
        public int UserId { get; set; }
        public decimal ShippingCost { get; set; }
        public decimal TotalCost { get; set; }
        public ShippingAddressDto ShippingAddressDto { get; set; }
        public PaymentMethodType PaymentMethodType { get; set; }
        public CreditCardDto CreditCardDto { get; set; }
        public CartItemDTO CartItemDTO { get; set; }
    }
}

File: Dokany UI/src/app/dtos/orderCreationDto.ts
import { ShippingAddressDto } from './shippingAddressDto';
import { CreditCardDto } from './creditCardDto';
import { CartItemDto } from './cartItemDto';

export class OrderCreationDto{
    Id: Number;
    ShippingId: Number;
    Description: String;
    UserId: Number;
    ShippingCost: Number;
    TotalCost: Number;
    PaymentMethod: Number;
    
    ShippingAddressDto: ShippingAddressDto;
    CreditCardDto: CreditCardDto;
    CartItemDto: CartItemDto;

    constructor(id: Number,
        shippingId: Number,
        description: String,
        userId: Number,
        shippingCost: Number,
        totalCost: Number,
        paymentMethod: Number,
        shippingAddressDto: ShippingAddressDto,
        creditCardDto: CreditCardDto,
        cartItemDto: CartItemDto){
            this.Id = id;
            this.ShippingId = shippingId;
            this.Description = description;
            this.UserId = userId;
            this.ShippingCost = shippingCost;
            this.TotalCost = totalCost;
            this.PaymentMethod = paymentMethod;
            this.ShippingAddressDto = shippingAddressDto;
            this.CreditCardDto = creditCardDto;
            this.CartItemDto = cartItemDto;
        }
}